app/src/main/java/es/wolfi/utils/PasswordGenerator.java

```diff
@@ -0,0 +1,142 @@
+package es.wolfi.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Iterator;
+import java.util.Random;
```

Comment:
Wold be better to use a cryptographically safe number generator like this: https://riptutorial.com/java/example/24855/generate-cryptographically-random-data

Reply:
changed to SecureRandom with a573e6b4ee82ca0854774161356bf6cb9c9da289